lib/Compiler.js

```diff
@@ -192,7 +194,12 @@ class Compiler {
 			this.cache.beginIdle();
 			this.running = false;
 
-			if (callback !== undefined) return callback(err, stats);
+			let callbackRst;
```

Comment:
You don’t need this. `SyncHook` returns nothing

Reply:
This `return` is for compatible with former behavior about the `finalCallback` return value. Has this useless before?

Reply:
yes, we tend to use `if(...) return callback(...)` as shortcut for `if(...) { callback(...); return; }`